app/exttarfile.py

```diff
@@ -0,0 +1,29 @@
+import io
+import tarfile
+import zstandard
+
+
+class ExtTarFile(tarfile.TarFile):
+    """Extends TarFile to support zstandard"""
+
+    @classmethod
+    def zstdopen(cls, name, mode="r", fileobj=None, cctx=None, dctx=None, **kwargs):  # type: ignore
+        """Open zstd compressed tar archive name for reading or writing.
+           Appending is not allowed.
+        """
+        if mode not in ("r"):
+            raise ValueError("mode must be 'r'")
+
+        try:
+            zobj = zstandard.open(fileobj or name, mode + "b", cctx=cctx, dctx=dctx)
+            with zobj:
+                data = zobj.read()
```

Comment:
Ugh.  Yeah,  you technically seek in your tarfile reads, as you get a list of all the members and then iterate that, rather than just iterating the tarfile object directly.

Reply:
the zstandard fobj raised when seeking backwards, what tarfile does a lot. it's all small enough.. *shrug*

Reply:
I'm pretty sure it wouldn't need to seek backwards if you changed `for info in tar.getmembers():` to `for info in tar:`, but it is small enough...